### PR TITLE
Template Datasource Functions

### PR DESCRIPTION
**Pull request by Krunal-Thakkar**
# Description
Template Datasource Functions

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Integration Test
image

- [ ] Unit Test
image

image

### CODE REVIEW

**Comment by Krunal-Thakkar** [comment]
@AkshaySainiDell @adarsh-dell  Please review it

**Comment by AkshaySainiDell** [comment]
@Krunal-Thakkar Could you please attach screenshot of the code coverage as well.

File: types/v1/types.go
Comment (AkshaySainiDell) [inline types/v1/types.go:1]: Update the copyright year to `2019 - 2024`

**Review by AkshaySainiDell — DISMISSED** [review]

File: types/v1/types.go
Comment (Krunal-Thakkar) [reply types/v1/types.go:1]: Updated

**Comment by adarsh-dell** [comment]
@Krunal-Thakkar , Can you please make sure that code coverage is >= 90%. Please run the UT and attach the % of coverage.

**Review by adarsh-dell — APPROVED** [review]

**Review by shenda1 — APPROVED** [review]

**Pull request merged**